pkg/file/pipeline/chunkfeeder_writer.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pipeline
+
+import (
+	"encoding/binary"
+)
+
+const span = 8
+
+type chunkFeeder struct {
+	size      int
+	next      ChainWriter
+	buffer    []byte
+	bufferIdx int
+}
+
+func NewChunkFeederWriter(size int, next ChainWriter) Interface {
```

Comment:
It would be very good to comment what is feeder writer and why it exists, as it is a very important component.